### ISSUE TITLE
基于可观测生态的 AIOps

### ISSUE THREAD

**Issue by QingyaFan** [QingyaFan/blog#68]
# 什么是AIOps

AIOps（Artificial Intelligence for IT Operations）是一种利用人工智能（AI）和机器学习（ML）技术来自动化和增强IT运营的方式。AIOps 主要用于监控、分析和优化复杂的IT系统和基础设施。通过收集和分析大量的日志、性能指标和事件数据，AIOps 能够实时检测和预测潜在的问题，自动生成解决方案，并减少人为干预。

个人认为 Ops 的三个阶段是 ManualOps、DevOps、AIOps：

- ManualOps，传统的手动运维
- DevOps：利用脚本或程序的形式来自动化
- AIOps：利用人工智能（AI）和机器学习（ML）技术来自动化

## AIOps 的核心功能

1. 数据收集和整合：从不同的IT系统和工具中收集数据，包括日志、性能指标、网络流量等，并将这些数据整合在一起；
2. 模式识别：利用机器学习算法识别数据中模式，从而发现异常和潜在的问题；
3. 自动化操作：基于检测到的问题，AIOps 系统可以自动执行操作，如告警通知、故障修复和资源调整等；
4. 预测分析：通过历史数据的分析，AIOps 可以预测未来可能发生的故障或性能问题，帮助企业提前预防；
5. 根因分析：当出现问题时，AIOps 系统能够快速分析并确定问题的根本原因，从而加快问题的解决过程。

## AIOps 的应用场景

AIOps 的应用场景广泛，包括自动化故障排除、性能优化、容量规划、安全监控等。随着IT系统的复杂性增加，AIOps 在提高运营效率、减少停机时间和提升用户体验方面变得越来越重要。

## AIOps 的相关开源项目

- 数据采集和存储：Prometheus（指标）、Elasticsearch（日志）、Fluentd（日志）

- AIOps的大脑（模型）：
    - [facebook prophet](https://github.com/facebook/prophet): Tool for producing high quality forecasts for time series data that has multiple seasonality with linear or non-linear growth.
    - [neural_prophet](https://github.com/ourownstory/neural_prophet): NeuralProphet is an easy to learn framework for interpretable time series forecasting. NeuralProphet is built on PyTorch and combines Neural Networks and traditional time-series algorithms, inspired by [Facebook Prophet](https://github.com/facebook/prophet) and [AR-Net](https://github.com/ourownstory/AR-Net).
    - [Python Outlier Detection (PyOD)](https://github.com/yzhao062/pyod): a versatile Python library for detecting anomalies in multivariate data. Whether you're tackling a small-scale project or large datasets, PyOD offers a range of algorithms to suit your needs.
    - [log-anomaly-detector](https://github.com/AICoE/log-anomaly-detector): It can connect to streaming sources and produce predictions of abnormal log lines. Internally it uses unsupervised machine learning. We incorporate a number of machine learning models to achieve this result. In addition it includes a human in the loop feedback system. **_public archived_**
    - [prometheus-anomaly-detector](https://github.com/AICoE/prometheus-anomaly-detector): This repository contains the prototype for a Prometheus Anomaly Detector (PAD) which can be deployed on OpenShift. The PAD is a framework to deploy a metric prediction model to detect anomalies in prometheus metrics.

- 自动化执行：Ansible
- 数据可视化：Grafana

# AIOps 与可观测

#57 里介绍了OpenTelemetry 生态采集、处理、导出可观测数据，这些数据一般会用于发现问题、问题定位，最终作为止损措施的依据，这里我们介绍一下基本的方法，以及如何将其智能化、自动化（AIOps）。

# 数据有哪些

todo：基于三维的故障传导模型，采集的应用、基础设施数据，没类数据都有多个维度

# 可观测生态中有哪些工具

todo：Prometheus/PromQL、Grafana等

# 发现问题

todo：检查热点：孤立森林？拓扑分析问题？结合做过的项目总结

# 定位问题

todo：多维度分析、相关性分析、大模型？

# 止损

todo：服务网格？

参考 
阿里云可观测智能化探索——从智能告警到利用LLM实现自然语言转PromQL [^1]

[^1]: https://mp.weixin.qq.com/s/Whizzzq2LzQXxzIgd7u7TQ
